### PR TITLE
Fix las 1.4 compatibility

### PR DESCRIPTION
**Pull request by taozywu**
Maybe there are some problems in the code, SO far I have not considered RGB and so on, but it can be compiled preliminarily. If you have any questions, please feel free to discuss! :)

### CODE REVIEW

**Comment by taozywu** [comment]
There is las 1.4 header.
image

There is las 1.2 header.
image

File: third_party/lasread/lidario.go [hunk 1]
```diff
@@ -385,8 +385,9 @@ func (las *LasFile) read() error {
 func (las *LasFile) readHeader() error {
 	las.Lock()
 	defer las.Unlock()
-	b := make([]byte, 243)
-	if _, err := las.f.ReadAt(b[0:243], 0); err != nil && err != io.EOF {
+	// For 1.4 set max length
+	b := make([]byte, 375)
```
Comment (mfbonfigli) [inline third_party/lasread/lidario.go:389]: While this should work for LAS 1.4 I suspect we might create compatibility problems with LAS 1.2. Should we first read the first 26 bytes up to the VersionMajor and VersionMinor and then decide how much more to read to fill the headers?

File: third_party/lasread/lidario.go [hunk 2]
```diff
@@ -479,8 +480,23 @@ func (las *LasFile) readHeader() error {
 	offset += 8
 	las.Header.MinZ = math.Float64frombits(binary.LittleEndian.Uint64(b[offset : offset+8]))
 	offset += 8
-	if las.Header.VersionMajor == 1 && las.Header.VersionMinor == 3 {
+	if las.Header.VersionMajor == 1 && las.Header.VersionMinor >= 3 {
```
Comment (mfbonfigli) [inline third_party/lasread/lidario.go:483]: The added headers are LAS 1.4 specific, they are not present in LAS 1.3 specs so we need an additional condition check here.

File: third_party/lasread/lidario.go [hunk 3]
```diff
@@ -479,8 +480,23 @@ func (las *LasFile) readHeader() error {
...
+		las.Header.NumberOfPointRecords = binary.LittleEndian.Uint64(b[offset : offset+8])
+		offset += 8
+		for i := 0; i < 15; i++ {
+			las.Header.NumberOfPointsByReturn[i] = binary.LittleEndian.Uint64(b[offset : offset+8])
+			offset += 8
+		}
+	}
+
+	if las.Header.NumberPoints < 1 {
+		las.Header.NumberPoints = int(las.Header.NumberOfPointRecords)
```
Comment (mfbonfigli) [inline third_party/lasread/lidario.go:499]: A bit unclear on why this is here.

File: third_party/lasread/tiler_las_reader.go [hunk 4]
```diff
@@ -50,21 +51,21 @@ func (lasFileLoader *LasFileLoader) readForOctree(inSrid int, eightBitColor bool
 		return err
 	}
 	if las.fileMode != "rh" {
-		recLengths := [4][4]int{{20, 18, 19, 17}, {28, 26, 27, 25}, {26, 24, 25, 23}, {34, 32, 33, 31}}
-
-		if las.Header.PointRecordLength == recLengths[las.Header.PointFormatID][0] {
-			las.usePointIntensity = true
```
Comment (mfbonfigli) [inline third_party/lasread/tiler_las_reader.go:56]: Unfortunately point intensity and point user data are still present but optional even in LAS 1.4. 
As such I don't think we can remove this block :/

**Comment by mfbonfigli** [comment]
Hello taozywu and thanks for your contribution! Please review the comments, I think that overall the direction is right but we need to make sure we don't have regressions in LAS format support.

File: third_party/lasread/lidario.go (hunk 1 above)
Comment (taozywu) [reply third_party/lasread/lidario.go:389]: I have tried both 1.2 and 1.4, but it is also ok to read the first 26 bytes to judge.I can check it later. Thank you  :)

File: third_party/lasread/lidario.go (hunk 3 above)
Comment (taozywu) [reply third_party/lasread/lidario.go:499]: I can verify it again. Sorry, I'm not familiar with GO.

File: third_party/lasread/lidario.go (hunk 1 above)
Comment (taozywu) [reply third_party/lasread/lidario.go:389]: image
----
this is las1.3 header, the total length is 235

File: third_party/lasread/lidario.go (hunk 2 above)
Comment (taozywu) [reply third_party/lasread/lidario.go:483]: According to the screenshot provided above, there is indeed a small problem with this piece, I will adjust it later. Besides, I don't have the LAS1.3 file at present, so I need to find the LAS1.3 file for testing.Please wait :)

**Pull request closed without merging**